### PR TITLE
doc improvement : applications: nrf_desktop: Document forced scan state

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces forced scan state description to the Bluetooth LE scan module documentation. Change also updates application README.

Jira: NCSDK-23219

### CODE REVIEW

**Comment by kapi-no** [comment]
Please provide a link to the PR in which the feature was added. I think we should have a process like this for the documentation follow-up PR, as it is easier to evaluate if the documentation PR covers the scope of the functional PR with code changes

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -537,7 +537,8 @@ The peripheral supports one wireless connection at a time, but it can be bonded
    See the :ref:`nrf_desktop_bluetooth_guide_fast_pair` documentation section for details.
 
 The nRF Desktop Bluetooth Central device scans for all bonded peripherals that are not connected.
-The scanning is interrupted when any device connected to the dongle through Bluetooth comes in use.
+Right after entering scanning state, the scanning operation is uninterruptable for predefined time (:kconfig:option:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S`) to speed up connection establishment with Bluetooth Peripherals.
```
Comment (kapi-no) [inline applications/nrf_desktop/README.rst:540]:
```suggestion
Right after entering the scanning state, the scanning operation is uninterruptable for a predefined time (:kconfig:option:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S`) to speed up connection establishment with Bluetooth Peripherals.
```

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -537,7 +537,8 @@ The peripheral supports one wireless connection at a time, but it can be bonded
...
+After the timeout, the scanning is interrupted when any device connected to the dongle through Bluetooth comes in use.
```
Comment (kapi-no) [inline applications/nrf_desktop/README.rst:541]: "comes in use" Can we be more specific here on what it means? HID reports exchange, etc.?

File: applications/nrf_desktop/doc/ble_scan.rst [hunk 3]
```diff
@@ -122,6 +124,25 @@ The scanning is interrupted if one of the following conditions occurs:
...
+If the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>` Kconfig option is set to a value greater than zero, the module enters the forced scan state after one of the following conditions:
+
+* boot or wakeup,
+* successful peripheral discovery,
+* peripheral disconnection,
+* connection failure,
+* bond erase,
+* scan request.
+
+After the time specified by the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>`, the module switches to regular scanning that can be interrupted by using connected peripherals and times out after the scan duration specified by :ref:`CONFIG_DESKTOP_BLE_SCAN_DURATION_S <config_desktop_app_options>` if there are peripherals connected over Bluetooth.
```
Comment (kapi-no) [inline applications/nrf_desktop/doc/ble_scan.rst:141]: I think we can break the sentence into two parts so it is simpler. Here is my suggestion 

```suggestion
After the time specified by the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>`, the module switches to regular scanning that can be interrupted by using connected peripherals.
The operation times out after the scan duration specified by :ref:`CONFIG_DESKTOP_BLE_SCAN_DURATION_S <config_desktop_app_options>` if there are peripherals connected over Bluetooth.
```

**Review by mkapala-nordic — APPROVED** [review]

File: applications/nrf_desktop/doc/ble_scan.rst (hunk 3 above)
Comment (divipillai) [inline applications/nrf_desktop/doc/ble_scan.rst:141]:
```suggestion
After the duration that is specified in the :ref:`CONFIG_DESKTOP_BLE_FORCED_SCAN_DURATION_S <config_desktop_app_options>` Kconfig option, the module switches to regular scanning.
The regular scanning can be interrupted if there are peripherals connected over Bluetooth and times out after the scan duration specified in the :ref:`CONFIG_DESKTOP_BLE_SCAN_DURATION_S <config_desktop_app_options>`.
```
Hope the meaning has not changed.

**Comment by MarekPieta** [comment]
> Please provide a link to the PR in which the feature was added. I think we should have a process like this for the documentation follow-up PR, as it is easier to evaluate if the documentation PR covers the scope of the functional PR with code changes

https://github.com/nrfconnect/sdk-nrf/pull/12107

File: applications/nrf_desktop/README.rst (hunk 2 above)
Comment (MarekPieta) [reply applications/nrf_desktop/README.rst:541]: ok, added clarification

File: applications/nrf_desktop/doc/ble_scan.rst (hunk 3 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/ble_scan.rst:141]: Rephrased.

**Review by kapi-no — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Pull request merged**